cookbooks/bcpc-hadoop/recipes/zabbix_web_monitoring.rb

```diff
@@ -0,0 +1,135 @@
+    end #if host_id.nil?
+
+    # iterate the web scenarios and create_or_update scenarios and related triggers and actions
+    scenarios = node.default[:bcpc][:hadoop][:zabbix][:web_monitoring][:scenarios] or {}
+    scenarios.each do | scenario |
+      Chef::Log.info("start processing web scenario #{scenario[:name]}")
+
+      # define names for the three parts
+      name_web     = "#{scenario[:name]}"
+      name_trigger = "#{scenario[:name]} trigger"
+      name_action  = "#{scenario[:name]} action"
+
+      # add scenario
+      zbx.httptests.create_or_update(
+        :name => name_web,
+        :delay => "15",
+        :retries=> "5",
+        :hostid => "#{host_id}",
+        :steps => [{
+          :no=>"1",
+          :name=> scenario[:name],
+          :url => scenario[:url],
+          :timeout=>"15",
+          :status_codes=>"200",
+          :follow_redirects=>"1",
+        }]
+      )
+
+      Chef::Log.info("added web scenario:" +
+        zbx.httptests.get_raw( { :filter => { :name => name_web } , :selectSteps => "extend" }).to_s)
+
+
+      # add triggers
+      trigger_expression = "{#{host_name}:#{scenario[:trigger_item]}[#{name_web}]." +
+        "#{scenario[:trigger_func]}}#{scenario[:trigger_cond]}"
+
+      zbx.triggers.create_or_update(
+        :description => name_trigger,
+        :expression => trigger_expression,
+        :comments => scenario[:description],
+        :hostid => host_id,
+        :priority => scenario[:severity],
+        :status => "0",
+        :type => "0"
+      )
+
+      Chef::Log.info("added trigger:" +
+        zbx.triggers.get_raw( { :filter => { :description => name_trigger }, :output => "extend", }).to_s)
+
+
+      # add action
+      action_status = node[:bcpc][:hadoop][:zabbix][:enable_alarming] ? 0 : 1 # status = 0 means enabled
+      esc_period = node[:bcpc][:hadoop][:zabbix][:escalation_period]
+      zbx.actions.create_or_update(
+        :name => name_action,
+        :eventsource => 0,
+        :status => action_status,
+        :esc_period => esc_period,
+        :filter => {
+          :evaltype => 1,
+          :conditions => [
+            {:conditiontype => 3, :operator => 2, :value => name_trigger},
+            {:conditiontype => 5, :operator => 0, :value => 1},
+            {:conditiontype => 16, :operator => 7, :value => ''}
+          ]
+        },
+        :operations => [{
+          :operationtype => 1,
+          :esc_step_from => 2,
+          :esc_step_to => 2,
+          :opcommand => {
+            :command => "#{node['bcpc']['zabbix']['scripts']['mail']}" +
+              " {TRIGGER.NAME} #{node.chef_environment}" +
+              " 2 '#{scenario[:description]}'" +
+              " #{host_name} admin",
+            :type => "0", :execute_on => "1"
+          },
+          :opcommand_hst => [:hostid => 0]
+        }]
+      )
+      Chef::Log.info("added action:" +
+        zbx.actions.get_raw( { :filter => { :name => "HBaseMasterAvailability_action" },
```

Comment:
For services like HBase, and all (even Graphite) please try to follow the pattern we use for copylogs where the service recipe updates the hash, so that each recipe is responsible for denoting the webmonitoring condition, rather than having a monolithic recipe of conditions. And it looks like this is just errantly running around.